code/modules/admin/sql_message_system.dm

```diff
@@ -219,6 +232,48 @@
 			browse_messages(target_ckey = target_ckey, agegate = TRUE)
 	qdel(query_find_edit_expiry_message)
 
+/proc/edit_message_severity(message_id, browse)
+	if(!SSdbcore.Connect())
+		to_chat(usr, "<span class='danger'>Failed to establish database connection.</span>")
+		return
+	message_id = text2num(message_id)
+	if(!message_id)
+		return	
+	var/datum/DBQuery/query_find_edit_note_severity = SSdbcore.NewQuery("SELECT type, targetckey, adminckey, severity FROM [format_table_name("messages")] WHERE id = [message_id] AND deleted = 0")
+	if(!query_find_edit_note_severity.warn_execute())
+		qdel(query_find_edit_note_severity)
+		return
+	if(query_find_edit_note_severity.NextRow())
+		var/list/severity_list = list("High Severity" = 4, "Medium Severity" = 3, "Minor Severity" = 2, "No Severity" = 1)
+		var/type = query_find_edit_note_severity.item[1]
+		var/target_ckey = query_find_edit_note_severity.item[2]
+		var/admin_ckey = query_find_edit_note_severity.item[3]
+		var/old_severity = query_find_edit_note_severity.item[4]
+		if(old_severity)
+			var/list/num2severity = list("4" = "high", "3" = "medium", "2" = "minor", "1" = "none") // there probably is a better way to do this - CitrusGender
+			old_severity = num2severity["[old_severity]"]
+		else
+			old_severity = "NA"
+		var/editor_ckey = sanitizeSQL(usr.ckey)
+		var/new_severity = input("Set the severity of the note.", "Severity", null, null) as null|anything in severity_list
+		if(!new_severity)
+			return
+		var/note_severity = severity_list[new_severity]
+		note_severity = sanitizeSQL(note_severity) // This might not be neccesary
+		var/edit_text = sanitizeSQL("Note severity edited by [editor_ckey] on [SQLtime()] from [old_severity] to [new_severity]<hr>")
+		var/datum/DBQuery/query_edit_note_severity = SSdbcore.NewQuery("UPDATE [format_table_name("messages")] SET severity = [note_severity], lasteditor = '[editor_ckey]', edits = CONCAT(IFNULL(edits,''),'[edit_text]') WHERE id = [message_id] AND deleted = 0")
+		if(!query_edit_note_severity.warn_execute())
+			qdel(query_edit_note_severity)
+			return
+		qdel(query_edit_note_severity)
+		log_admin_private("[key_name(usr)] has edited the severity of a [type] [(type == "note" || type == "message" || type == "watchlist entry") ? " for [target_ckey]" : ""] made by [admin_ckey] from [old_severity] to [new_severity]")
+		message_admins("[key_name_admin(usr)] has edited the severity time of a [type] [(type == "note" || type == "message" || type == "watchlist entry") ? " for [target_ckey]" : ""] made by [admin_ckey] from [old_severity] to [new_severity]")
```

Comment:
Replace `key_name` and `key_name_admin` with a cached var like https://github.com/tgstation/tgstation/blob/master/code/modules/admin/sql_message_system.dm#L234-L253
This is to avoid nasties if usr gets deleted while the query is running.

Reply:
Also use `key` not `ckey`.